command-line/lib/Command/Key/Hash.hs

```diff
@@ -0,0 +1,53 @@
+{-# LANGUAGE FlexibleContexts #-}
+{-# LANGUAGE NamedFieldPuns #-}
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE QuasiQuotes #-}
+
+{-# OPTIONS_HADDOCK hide #-}
+
+module Command.Key.Hash
+    ( Cmd (..)
+    , mod
+    , run
+    ) where
+
+import Prelude hiding
+    ( mod )
+
+import Cardano.Address.Style.Shelley
+    ( liftXPub )
+import Cardano.Multisig
+    ( KeyHash (..), hashKey )
+import Codec.Binary.Bech32.TH
+    ( humanReadablePart )
+import Options.Applicative
+    ( CommandFields, Mod, command, footerDoc, helper, info, progDesc )
+import Options.Applicative.Encoding
+    ( Encoding, encodingOpt )
+import Options.Applicative.Help.Pretty
+    ( string )
+import System.IO
+    ( stdin, stdout )
+import System.IO.Extra
+    ( hGetXPub, hPutBytes )
+
+newtype Cmd = Hash
+    { encoding :: Encoding
+    } deriving (Show)
+
+mod :: (Cmd -> parent) -> Mod CommandFields parent
+mod liftCmd = command "hash" $
+    info (helper <*> fmap liftCmd parser) $ mempty
+        <> progDesc "Get the hash of a public key"
+        <> footerDoc (Just $ string $ mconcat
+            [ "The public key is read from stdin."
+            ])
+  where
+    parser = Hash
+        <$> encodingOpt [humanReadablePart|xpub|]
```

Comment:
I don't think `xpub` is the correct HRP for this.
Perhaps `scripthash` or something?

Reply:
indeed! change to `script_hash` here, also changed the default hrp for public key hashes to `xpub_hash` - see https://github.com/input-output-hk/cardano-addresses/pull/76/commits/6fb551f7bfef35387c49c64b9a3fbf17f520c338